types/config.go

```diff
@@ -146,6 +155,15 @@ type ProviderConfig struct {
 	// you can use to pass role information to an EC2 instance when the instance starts.
 	InstanceProfile string
 
+	// Machine specifies the type of machine (pc or q35) (Used only for ProxMox yet)
+	Machine string `cloud:"machine"`
```

Comment:
The ProviderConfig struct already has a "Flavor" member, which is used to specify a machine type for other cloud providers (such as for example aws and gcp). It seems to me you could use that instead of adding a new field to the struct.

Reply:
> The ProviderConfig struct already has a "Flavor" member, which is used to specify a machine type for other cloud providers (such as for example aws and gcp). It seems to me you could use that instead of adding a new field to the struct.

Maybe for Proxmox, in general, transfer all fields to ProxmoxConfig, and not use CloudConfig at all?

         "ImageName": "mytest",
         "Arch": "x86_64",
         "Machine": "q35",
         "Sockets: 2
         "Cores: 4
         "Numa": true
         "Memory": "4096M"

It just seems to me that if you add settings in the course of development, then you can add a bunch of additional settings to any providers. And in ProxmoxConfig, I have displayed far from everything that is configured there via Api.